firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/TransformMutation.java

```diff
@@ -207,7 +211,12 @@ private Document requireDocument(@Nullable MaybeDocument maybeDoc) {
         previousValue = ((Document) baseDoc).getField(fieldTransform.getFieldPath());
       }
 
-      transformResults.add(transform.applyToLocalView(previousValue, localWriteTime));
+      FieldValue currentValue = null;
```

Comment:
It seems as if we could use currentValue if available or fall back on previousValue if not currentValue is not available. At that point the signatures for the TransformOperations wouldn't have to change.

Reply:
Done